addons/mass_mailing_sale/models/mailing_mailing.py

```diff
@@ -10,52 +10,71 @@ class MassMailing(models.Model):
     _name = 'mailing.mailing'
     _inherit = 'mailing.mailing'
 
-    sale_quotation_count = fields.Integer('Quotation Count', groups='sales_team.group_sale_salesman', compute='_compute_sale_quotation_count')
-    sale_invoiced_amount = fields.Integer('Invoiced Amount', groups='sales_team.group_sale_salesman', compute='_compute_sale_invoiced_amount')
+    sale_quotation_count = fields.Integer('Quotation Count', compute='_compute_sale_quotation_count')
+    sale_invoiced_amount = fields.Integer('Invoiced Amount', compute='_compute_sale_invoiced_amount')
 
     @api.depends('mailing_domain')
     def _compute_sale_quotation_count(self):
-        has_so_access = self.env['sale.order'].check_access_rights('read', raise_exception=False)
-        if not has_so_access:
-            self.sale_quotation_count = 0
-            return
         for mass_mailing in self:
-            mass_mailing.sale_quotation_count = self.env['sale.order'].search_count(mass_mailing._get_sale_utm_domain())
+            mass_mailing.sale_quotation_count = self.env['sale.order'].sudo().search_count(
+                [('source_id', '=', mass_mailing.source_id.id)]
+            )
 
     @api.depends('mailing_domain')
     def _compute_sale_invoiced_amount(self):
-        if not self.user_has_groups('sales_team.group_sale_salesman') or not self.user_has_groups('account.group_account_invoice'):
-            self.sale_invoiced_amount = 0
-            return
         for mass_mailing in self:
             domain = expression.AND([
-                mass_mailing._get_sale_utm_domain(),
+                [('source_id', '=', mass_mailing.source_id.id)],
                 [('state', 'not in', ['draft', 'cancel'])]
             ])
-            moves = self.env['account.move'].search_read(domain, ['amount_untaxed_signed'])
+            moves = self.env['account.move'].sudo().search_read(domain, ['amount_untaxed_signed'])
```

Comment:
You could probably make a search_read for all mailings (source_id, in, self.source_id.ids). Then in the loop, filter the relevant account moves and sum their untaxed amount. Would allow to have a single search for all mailings to display.

Reply:
Like this?

```py
def _compute_sale_invoiced_amount(self):
    domain = expression.AND([
        [('source_id', 'in', self.source_id.ids)],
        [('state', 'not in', ['draft', 'cancel'])]
    ])
    moves_data = self.env['account.move'].sudo().read_group(
        domain, ['source_id', 'amount_untaxed_signed'], ['source_id'],
    )
    mapped_data = {datum['source_id'][0]: datum['amount_untaxed_signed'] for datum in moves_data}
    for mass_mailing in self:
        mass_mailing.sale_invoiced_amount = mapped_data.get(mass_mailing.source_id.id, 0)
```

Reply:
It is, I checked :)